src/compass/utils/helpers.py

```diff
@@ -340,3 +341,55 @@ def open_raster(filename, band=1):
     raster = ds.GetRasterBand(band).ReadAsArray()
 
     return raster
+
+
+def write_raster(filename, data_list, descriptions,
+                 data_type=gdal.GDT_Float32, data_format='ENVI'):
+    '''
+    Write a multiband GDAL-friendly raster to disk.
+    Each dataset allocated in the output file contains
+    a description of the dataset allocated for that band
+
+    Parameters
+    ----------
+    filename: str
+        File path where to store output dataset
+    data_list: list[np.ndarray]
+        List of numpy.ndarray to allocate for each
+        raster band. All datasets within the list
+        are assumed to have the same shape
+    descriptions: list[str]
+        List of strings containing a description
+        for the bands to allocate
+    data_type: gdal.dtype
+        GDAL dataset type
+    format: gdal.Format
+        Format for GDAL output file
+    '''
+
+    error_channel = journal.error('helpers.write_raster')
+
+    # Check number of datasets match number of descriptions
+    if len(data_list) != len(descriptions):
+        err_str = f'Number of datasets to write does not match' \
+                  f'the number of descriptions ' \
+                  f'{len(data_list)} != {len(descriptions)}'
+        error_channel.log(err_str)
+        raise ValueError(err_str)
```

Comment:
Not sure if we want to change these since we're in the process of converting all logging to python, but Michael pointed out that the `raise ...` statements in COMPASS that occur after an `error_channel.log` call will never be reached. The error channel is "fatal" aka it raises it's own error first:

```python
In [2]: import journal
In [3]: c = journal.error('test1')

In [5]: c.log("bad")
journal: bad
---------------------------------------------------------------------------
ApplicationError                          Traceback (most recent call last)
Cell In[5], line 1
----> 1 c.log("bad")

ApplicationError: test1: application error
```